### PR TITLE
Gradle 8 upgrades

### PR DESCRIPTION
**Pull request by pepone**
Upgrades java builds to use Gradle 8.x

### CODE REVIEW

File: java/test/android/controller/build.gradle [hunk 1]
```diff
@@ -51,11 +51,11 @@ repositories {
 }
 
 android {
-    compileSdkVersion 33
+    compileSdk 33
 
     defaultConfig {
         applicationId "com.zeroc.testcontroller"
-        minSdkVersion 31
+        minSdkVersion 24
```
Comment (externl) [inline java/test/android/controller/build.gradle:58]: why decrease this one?

**Review by externl — APPROVED** [review]

File: java/test/android/controller/build.gradle (hunk 1 above)
Comment (pepone) [reply java/test/android/controller/build.gradle:58]: This is the version we support, as per the release notes, and I wanted to test it.

File: java-compat/test/android/controller/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```
Comment (bernardnormier) [inline java-compat/test/android/controller/gradle/wrapper/gradle-wrapper.properties:3]: Should this be 8.3?

File: java-compat/test/android/controller/build.gradle [hunk 3]
```diff
@@ -10,7 +10,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.4.2'
+        classpath 'com.android.tools.build:gradle:8.1.1'
```
Comment (bernardnormier) [inline java-compat/test/android/controller/build.gradle:13]: I don't follow: the distributionUrl is gradle-8.3 but the classpath is 8.1.1? Why this discrepancy?

Comment (pepone) [reply java-compat/test/android/controller/build.gradle:13]: They are different packages, one is Gradle and the other is the Android plug-in for Gradle

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**